File: solutions/r/proverb/1/proverb.R
recite <- function(wanted) {
  if (length(wanted) == 0) {
    return(c())
  }

  first_lines <- if (length(wanted) == 1) {
    c()
  } else {
    paste0(
      "For want of a ",
      wanted[-length(wanted)],
      " the ",
      wanted[-1],
      " was lost."
    )
  }

  c(
    first_lines,
    paste0("And all for the want of a ", wanted[1], ".")
  )
}

File: solutions/vbnet/proverb/1/Proverb.vb
Imports System.Linq

Public Module Proverb
    Public Function Recite(ByVal subjects As String()) As String()
        If subjects.Length = 0 Then
            Return New String(-1) {}
        End If

        Return subjects.Zip(subjects.Skip(1), Function(a, b) $"For want of a {a} the {b} was lost.").Append($"And all for the want of a {subjects(0)}.").ToArray()
    End Function
End Module